agent/httpclient/httpclient.go

```diff
@@ -67,8 +68,11 @@ func New(timeout time.Duration, insecureSkipVerify bool) *http.Client {
 		}).Dial,
 		TLSHandshakeTimeout: 10 * time.Second,
 	}
+
+	transport.TLSClientConfig = &tls.Config{}
+	cipher.WithSupportedCipherSuites(transport.TLSClientConfig)
 	if insecureSkipVerify {
-		transport.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+		transport.TLSClientConfig.InsecureSkipVerify = true
```

Comment:
You don't need a branch here:

`transport.TLSClientConfig.InsecureSkipVerify = insecureSkipVerify`